### PR TITLE
LPS-148316 Allow defining friendlyURL for documents when staging is enabled

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-148316 when exporting a file staging export the  the URLFriendlyEntries
- LPS-148316 when importing a file on staging update the URLFriendlyEntries on the live file with the staging model has.
- LPS-148316 update the FriendlyURLEntries to match the history of imported/exported entries
- LPS-148316 override the deleteFileEntry with the lower call to let the staging model access to it
- LPS-148316 test the behavior of the staging destaging for the friendlyURLEntries

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/exportimport/data/handler/FileEntryStagedModelDataHandler.java [hunk 1]
```diff
@@ -329,7 +333,7 @@ protected void doExportStagedModel(
 		}
 
 		_exportMetaData(portletDataContext, fileEntryElement, fileEntry);
-
+		_exportFriendlyURLEntries(portletDataContext, fileEntry);
 		_exportAssetDisplayPage(portletDataContext, fileEntry);
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/exportimport/data/handler/FileEntryStagedModelDataHandler.java:337]: Unless there's a dependency order, sort these calls in the same way everywhere. See my other comment about this for the other occurence.

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/exportimport/data/handler/FileEntryStagedModelDataHandler.java [hunk 2]
```diff
@@ -672,6 +676,9 @@ protected void doImportStagedModel(
 
 			_importAssetDisplayPage(
 				portletDataContext, fileEntry, importedFileEntry);
+			_importFriendlyURLEntries(
+				portletDataContext, fileEntry, importedFileEntry,
+				serviceContext);
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/exportimport/data/handler/FileEntryStagedModelDataHandler.java:681]: Sort these calls in the same way as in the other place.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/exportimport/data/handler/FileEntryStagedModelDataHandler.java [hunk 3]
```diff
@@ -987,6 +1013,55 @@ private void _importAssetDisplayPage(
...
+				existingFriendlyURLEntry.setClassPK(
+					importedFileEntry.getFileEntryId());
+
+				_friendlyURLEntryLocalService.updateFriendlyURLEntry(
+					existingFriendlyURLEntry);
+			}
+		}
+
+		_updateFriendlyURLEntries(fileEntry, importedFileEntry, serviceContext);
+	}
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/exportimport/data/handler/FileEntryStagedModelDataHandler.java:1063]: Thats the question that I wanted to make you about the staging. 
I could find the way to make it possible.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Pull request closed without merging**